tfhe/src/shortint/server_key/tests/noise_level.rs

```diff
@@ -0,0 +1,31 @@
+use crate::shortint::ciphertext::NoiseLevel;
+use crate::shortint::keycache::KEY_CACHE;
+use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+
+#[test]
+fn test_noise_level_propagation() {
```

Comment:
one function per operator being tested please, with corner cases (i.e. 0 and other values for scalars)

Reply:
and when I say per operator, it includes, default, checked, smart and unchecked with assign versions to be sure we cover all of the paths

Reply:
except smart versions because of the mutable references potentially but it would automate most of the rest to check all paths